todo/src/routes/routes.ts

```diff
@@ -0,0 +1,133 @@
+import { server } from "../server"
+import { ResponseToolkit, Request } from "@hapi/hapi"
+import { tasks } from "../models";
+
+export function getRoute() {
+
+    // GET all Tasks
+    server.route({
+        method: 'GET',
+        path: '/task',
+        handler: (request: Request, h: ResponseToolkit) => {
+            return tasks;
+        }
+    });
+
+    // Only grab the todo that has the completed status of true
+    server.route({
+        method: 'GET',
+        path: '/task/completed',
+        handler: (request: Request, h: ResponseToolkit) => {
+            const found = tasks.some(task => true === task.complete);
+            if(found) {
+                return(tasks.filter(task => true === task.complete))
+            } else {
+                return (`No todo is completed`);
+            }
+        }
+    });
+
+    // Only grab the tasks that have a completed status of false
+    server.route({
+        method: 'GET',
+        path: '/task/incomplete',
+        handler: (request: Request, h: ResponseToolkit) => {
+            const found = tasks.some(task => false === task.complete);
+            if(found) {
+                return(tasks.filter(task => false === task.complete))
+            } else {
+                return (`No todo is completed`);
+            }
+        }
+    });
+
+    // Sorting the ID numbers
+    server.route({
+        method: 'GET',
+        path: '/task/sorted',
+        handler: (request: Request, h: ResponseToolkit) => {
+            var sortedTasks = tasks;
+            sortedTasks.sort((task1, task2) => task1.id - task2.id);
+           return sortedTasks;
+        }
+    });
+
+    // GET a single task
+    server.route({
+        method: 'GET',
+        path: '/task/{id}',
+        handler:(request: Request, h: ResponseToolkit) => { 
+            console.log(request.params);
+            console.log(request.params.id);
+
+            const found = tasks.some(task => task.id === parseInt(request.params.id));
+
+            if(found) {
+                return(tasks.filter(task => task.id === parseInt(request.params.id)));
+            } else {
+                return (`No task with the ID of ${request.params.id}`)
+            }
+            //return JSON.stringify(models[request.params.id]);  
+        }
+    });
+
+    type NewTask = {
+        id: number;
+        todo: string;
+        dueDate: string;
+        complete: boolean;
+    }
```

Comment:
We are using a type alias here.

1.) why is this in the routes file when it could be in the models and used to give stronger typing to the models?
2.) this could probably just be an interface instead. A good rule of them is, if you don't know why something should be a type in typescript it can probably just be an interface instead

Reply:
I turn this into an interface in the next PR but I'll go back and implement it here